firmware/Sources/MID/MID_DABS/mid_dabs.c

```diff
@@ -0,0 +1,292 @@
+	blink_mode_wait = 0x01U,
+	blink_mode_cc_chg = 0x02U,
+	blink_mode_cc_dchg = 0x03U,
+	blink_mode_cv_chg = 0x04U,
+	blink_mode_cv_dchg = 0x05U,
+	blink_mode_cp_chg = 0x06U,
+	blink_mode_cp_dchg = 0x07U,
+	blink_mode_err_hsvolt = 0x08U,
+	blink_mode_err_lsvolt = 0x09U,
+	blink_mode_err_lscurr = 0x0AU,
+	blink_mode_err_comm = 0x0BU,
+	blink_mode_err_temp = 0x0CU,
+	blink_mode_err_int = 0x0DU
+}blink_mode_e;
+
+typedef struct{
+	blink_mode_e mode; 	//Blinking mode
+	uint8_t steps;		//Number of steps of the mode
+	uint8_t ledsStep[5];//Array with the output of the leds in each step
+}blink_conf_s;
+
+typedef struct{
+	blink_conf_s mode;		// Actual mode of the leds
+	uint8_t step;			// Actual step of the mode
+	blink_conf_s prevMode;	//Previous mode of the leds
+}blink_s;
+/**********************************************************************************/
+/*                      Definition of exported constant data                      */
+/**********************************************************************************/
+
+/**********************************************************************************/
+/*                    Declaration of local function prototypes                    */
+/**********************************************************************************/
+static MID_DABS_result_e CheckBlinkStatus(MID_REG_mode_e epcmode, int16_t lscurr,
+		MID_REG_errorStatus_s * errors, blink_conf_s * mode);
+
+static MID_DABS_result_e SetLeds(uint8_t state);
+/**********************************************************************************/
+/*                       Definition of local constant data                        */
+/**********************************************************************************/
+// Modes the leds can be
+const blink_conf_s idleMode = {
+	blink_mode_idle, 2,	{0x00,0x0F,0x00,0x00,0x00}
+};
+const blink_conf_s waitMode = {
+	blink_mode_wait, 3,	{0x00,0x0A, 0x05,0x00,0x00}
+};
+const blink_conf_s ccChgMode = {
+	blink_mode_cc_chg, 5, {0x00,0x08,0x04,0x02,0x01}
+};
+const blink_conf_s ccDchgMode = {
+	blink_mode_cc_dchg, 5,	{0x00,0x01,0x02,0x04,0x08}
+};
+const blink_conf_s cvChgMode = {
+	blink_mode_cv_chg, 5, {0x0F,0x0E,0x0D,0x0B,0x07}
+};
+const blink_conf_s cvDchgMode = {
+	blink_mode_cv_dchg, 5,	{0x0F,0x07,0x0B,0x0D,0x0E}
+};
+const blink_conf_s cpChgMode = {
+	blink_mode_cp_chg, 5,	{0x00,0x01,0x03,0x07,0x0F}
+};
+const blink_conf_s cpDchgMode = {
+	blink_mode_cp_dchg, 5,	{0x00,0x08,0x0C,0x0E,0x0F}
+};
+const blink_conf_s errHsvoltMode = {
+	blink_mode_err_hsvolt, 1,	{0x01,0x00,0x00,0x00,0x00}
+};
+const blink_conf_s errlsvoltMode = {
+	blink_mode_err_lsvolt, 1,	{0x08,0x00,0x00,0x00,0x00}
+};
+const blink_conf_s errlscurrMode = {
+	blink_mode_err_lscurr, 1, {0xC,0x00,0x00,0x00,0x00}
+};
+const blink_conf_s errCommMode = {
+	blink_mode_err_comm, 1,	{0x0A,0x00,0x00,0x00,0x00}
+};
+const blink_conf_s errTempMode = {
+	blink_mode_err_temp, 1,	{0x03,0x00,0x00,0x00,0x00}
+};
+const blink_conf_s errIntMode = {
```

Comment:
psychopat!!!

0b00001010... Please, use binary

Reply:
Proud of it, so don´t mess with me :)

Update it with the binary commands as a comment

Reply:
As it works, I respect it.